logging/impl.go

```diff
@@ -32,6 +32,11 @@ type (
 		zapcore.Entry
 		fields []zapcore.Field
 	}
+
+	implWith struct {
```

Comment:
[q] Why not add a `logFields` field to `impl` and just create a new `impl` object when `WithFields` is called? Or, are we relying on chains of derived `implWith`s to point back to the same `*impl` object and therefore have the same logging level that can be configured by one `log` pattern?

Reply:
> Or, are we relying on chains of derived implWiths to point back to the same *impl object and therefore have the same logging level that can be configured by one log pattern?

yup exactly this! if we return a new `*impl` object we currently have no way of keeping track of it without overwriting the registry entry for the original logger